WooCommerce/WooCommerceTests/Tools/InfiniteScroll/PaginationTrackerTests.swift

```diff
@@ -0,0 +1,120 @@
+import XCTest
+@testable import WooCommerce
+
+final class PaginationTrackerTests: XCTestCase {
+    private let paginationTracker = PaginationTracker(pageFirstIndex: 0, pageSize: 30)
+    private let mockDelegate = MockPaginationTrackerDelegate()
+
+    override func setUp() {
+        super.setUp()
+        paginationTracker.delegate = mockDelegate
+    }
+
+    func test_ensuring_next_page_after_syncing_first_page_triggers_delegate_to_sync_two_pages() {
+        // Arrange
+        var syncedPages = [PageInformation]()
+        mockDelegate.onSync = { pageNumber, pageSize, reason, onCompletion in
+            syncedPages.append(PageInformation(pageNumber: pageNumber, pageSize: pageSize, reason: reason))
+            onCompletion?(.success(true))
+        }
+
+        // Action
+        paginationTracker.syncFirstPage()
+        paginationTracker.ensureNextPageIsSynced()
+
+        // Assert
+        XCTAssertEqual(syncedPages, [
+            PageInformation(pageNumber: 0, pageSize: 30, reason: nil),
+            PageInformation(pageNumber: 1, pageSize: 30, reason: nil)
+        ])
```

Comment:
OMG I just tried this out of curiosity and it works!

```swift
        XCTAssertEqual(syncedPages, [
            .init(pageNumber: 0, pageSize: 30, reason: nil),
            .init(pageNumber: 1, pageSize: 30, reason: nil)
        ])
```

TIL

Reply:
niice! it does shorten the code a lot, updated in https://github.com/woocommerce/woocommerce-ios/pull/2750/commits/e196384c52d2f0549c001ef79be8b9fc2f3581bc